### PR TITLE
Align the strings keys between platforms

### PR DESCRIPTION
**Pull request by mionescu95**
## JIRA
[MOB-3665](https://karhoo.atlassian.net/browse/MOB-3665)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: KarhooUISDK/CommonUI/UITexts.swift [hunk 1]
```diff
@@ -14,462 +14,446 @@ public enum UITexts {
...
+        public static let locale = "kh_uisdk_locale".localized
+        public static let checked = "kh_uisdk_checked".localized
+        public static let unchecked = "kh_uisdk_unchecked".localized
+        public static let checkout = "kh_uisdk_booking_checkout_title".localized
     }
 
     public enum QuoteCell {
-        public static let details = "Text.QuoteCell.Details".localized
-        public static let driverArrival = "Text.QuoteCell.DriverArrival".localized
+        public static let details = "kh_uisdk_quoteCell_details".localized
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/UITexts.swift:57]: quote_cell* not quoteCell

File: KarhooUISDK/CommonUI/UITexts.swift [hunk 2]
```diff
@@ -14,462 +14,446 @@ public enum UITexts {
...
-        public static let mobilePhone = "Text.User.MobilePhoneNumber".localized
-        public static let signupEmailSubscribeCopy = "Text.User.EmailSubscription".localized
-        public static let signupPendingMessage = "Text.User.SignupPendingMessage".localized
-        public static let forgottenPassword = "Text.User.ForgottenPassword".localized
+        public static let firstName = "kh_uisdk_first_name_hint".localized
+        public static let lastName = "kh_uisdk_last_name_hint".localized
+        public static let email = "kh_uisdk_email_hint".localized
+        public static let mobilePhone = "kh_uisdk_mobile_phone_number_hint".localized
+        public static let signupEmailSubscribeCopy = "kh_uisdk_signupEmailSubscribeCopy".localized
+        public static let signupPendingMessage = "kh_uisdk_signupPendingMessage".localized
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/UITexts.swift:92]: key does not conform to standard

File: KarhooUISDK/CommonUI/UITexts.swift [hunk 3]
```diff
@@ -14,462 +14,446 @@ public enum UITexts {
...
+        public static let forgottenPassword = "kh_uisdk_forgottenPassword".localized
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/UITexts.swift:93]: key does not conform to standard

File: KarhooUISDK/CommonUI/UITexts.swift [hunk 4]
```diff
@@ -14,462 +14,446 @@ public enum UITexts {
...
-        public static let missingPaymentSDKToken = "Text.Errors.failedToInitialisePaymentSetup".localized
-        public static let noResultsFound = "Text.Errors.NoResultsFound".localized
-        public static let missingPhoneNumber = "Text.Errors.MissingPhoneNumber".localized
-        public static let invalidPhoneNumber = "Text.Errors.InvalidPhoneNumber".localized
-        public static let insufficientBalanceForLoyaltyBurning = "Text.Errors.InsufficientBalanceForLoyaltyBurning".localized
-        public static let unsupportedCurrency = "Text.Error.UnsupportedCurrency".localized
-        public static let unknownLoyaltyError = "Text.Error.UnkownLoyaltyError".localized
-        public static let loyaltyModeNotEligibleForPreAuth = "Text.Error.LoyaltyModeNotEligibleForPreAuth".localized
+        public static let flightNumberValidatorError = "kh_uisdk_errors_flightNumberValidatorError".localized
+        public static let noDetailsAvailable = "kh_uisdk_something_went_wrong".localized
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/UITexts.swift:100]: Same key as `somethingWentWrong` below. It had a dedicated error text before.

File: KarhooUISDK/CommonUI/UITexts.swift [hunk 5]
```diff
@@ -14,462 +14,446 @@ public enum UITexts {
...
+        public static let somethingWentWrong = "kh_uisdk_something_went_wrong".localized
+        public static let cantSendEmail = "kh_uisdk_errors_cantSendEmail".localized
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/UITexts.swift:102]: key does not conform to standard

File: KarhooUISDK/CommonUI/UITexts.swift [hunk 6]
```diff
@@ -14,462 +14,446 @@ public enum UITexts {
...
+        public static let getUserFail = "kh_uisdk_errors_getUserFail".localized
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/UITexts.swift:103]: key does not conform to standard

Comment (dianakarh) [reply KarhooUISDK/CommonUI/UITexts.swift:103]: not going to add this comment to all strings as they are too many. just please check all and make sure they conform to the agreed upon standard

File: KarhooUISDK/CommonUI/UITexts.swift [hunk 7]
```diff
@@ -14,462 +14,446 @@ public enum UITexts {
...
     /* Payment Error */
     public enum PaymentError {
-        public static let paymentAlertTitle = "Text.Error.Payment.Alert.Title".localized
-        public static let paymentAlertMessage = "Text.Error.Payment.Alert.Message".localized
-        public static let paymentAlertUpdateCard = "Text.Error.Payment.Alert.ButtonUpdateCard".localized
-        public static let noDetailsMessage = "Text.Error.Payment.NoDetails.Message".localized
+        public static let paymentAlertTitle = "kh_uisdk_payment_issue".localized
+        public static let paymentAlertMessage = "kh_uisdk_payment_issue_message".localized
+        public static let paymentAlertUpdateCard = "kh_uisdk_payment_paymentAlertUpdateCard".localized
+        public static let noDetailsMessage = "kh_uisdk_something_went_wrong".localized
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/UITexts.swift:137]: Wrong key mapped

File: KarhooUISDK/CommonUI/UITexts.swift [hunk 8]
```diff
@@ -14,462 +14,446 @@ public enum UITexts {
...
+        public static let noLocationPermissionConfirm = "kh_uisdk_settings".localized
     }
 
     public enum Availability {
-        public static let allCategory = "Text.Availability.AllCategory".localized
-        public static let noQuotesInSelectedCategory = "Text.Availability.NoQuotesInSelectedCategory".localized
-        public static let noQuotesForSelectedParameters = "Text.Availability.NoQuotesForSelectedParameters".localized
+        public static let allCategory = "kh_uisdk_filter_all".localized
+        public static let noQuotesInSelectedCategory = "kh_uisdk_noQuotesInSelectedCategory".localized
+        public static let noQuotesForSelectedParameters = "kh_uisdk_K3002".localized
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/UITexts.swift:246]: Looks like a wrong key was mapped. Please double check and ignore the comment if the mapping is correct

File: KarhooUISDK/CommonUI/UITexts.swift [hunk 9]
```diff
@@ -14,462 +14,446 @@ public enum UITexts {
...
-        public static let fleetCapabilities = "Text.Quotes.FilterFleetCapabilities".localized
-        public static let quoteTypes = "Text.Quotes.FilterQuoteTypes".localized
-        public static let serviceAgreements = "Text.Quotes.FilterServiceAgreements".localized
+        public static let vehicleType = "kh_uisdk_filter_vehicle_types".localized
+        public static let vehicleClass = "kh_uisdk_filter_vehicle_class".localized
+        public static let vehicleExtras = "kh_uisdk_filter_vehicle_extras".localized
+        public static let ecoFriendly = "kh_uisdk_filter_eco_friendly".localized
+        public static let fleetCapabilities = "kh_uisdk_filter_fleet_capabilities".localized
+        public static let quoteTypes = "kh_uisdk_filter_quote_types".localized
+        public static let serviceAgreements = "kh_uisdk_filter_cancellation_and_waiting_time".localized
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/UITexts.swift:282]: can we change the key name to `kh_uisdk_filter_service_agreements`? I'd keep it more generic than limiting to cancellation and waiting time.

File: KarhooUISDK/CommonUI/UITexts.swift [hunk 10]
```diff
@@ -14,462 +14,446 @@ public enum UITexts {
...
+        public static let tripCancelBookingConfirmationAlertTitle = "kh_uisdk_cancel_your_ride".localized
+        public static let tripCancelBookingFailedAlertTitle = "kh_uisdk_difficulties_cancelling_title".localized
+        public static let tripCancelBookingFailedAlertMessage = "kh_uisdk_difficulties_cancelling_message".localized
+        public static let tripCancelBookingFailedAlertCallFleetButton = "kh_uisdk_contact_fleet".localized
+        public static let tripCancellingActivityIndicatorText = "kh_uisdk_cancelling_ride".localized
+        public static let tripContactDriver = "kh_uisdk_contact_driver".localized
+        public static let tripContactFleet = "kh_uisdk_contact_fleet".localized
+        public static let tripRideOptions = "kh_uisdk_tripRideOptions".localized
+        public static let noDriversAvailableTitle = "kh_uisdk_booking_failed".localized
+        public static let noDriversAvailableMessage = "kh_uisdk_booking_failed_body_ios".localized
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/UITexts.swift:349]: This key doesn't look representative for the error. Same for the title above it

File: KarhooUISDK/CommonUI/UITexts.swift [hunk 11]
```diff
@@ -14,462 +14,446 @@ public enum UITexts {
...
+        static let bookingMessage = "kh_uisdk_airport_bookingMessage".localized
     }
 
     public enum AddressScreen {
-        static let noRecentAddresses = "Text.Address.NoRecentAddress".localized
-        static let noResultsFound = "Text.Errors.NoResultsFound".localized
-        static let setOnMap = "Text.Address.SetOnMap".localized
-        static let currentLocation = "Text.Address.CurrentLocation".localized
+        static let noRecentAddresses = "kh_uisdk_recents_empty".localized
+        static let noResultsFound = "kh_uisdk_something_went_wrong".localized
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/UITexts.swift:423]: The key seems a bit too generic.

File: KarhooUISDK/CommonUI/UITexts.swift [hunk 12]
```diff
@@ -14,462 +14,446 @@ public enum UITexts {
...
-        static let resetFilter = "Text.Quotes.ResetFilter".localized
-        static let filtersPassengers = "Text.Quotes.FiltersPassengers".localized
-        static let filtersLuggages = "Text.Quotes.FiltersLuggages".localized
+        static let freeCancellation = "kh_uisdk_filter_free_cancellation".localized
+        static let freeCancellationPrebook = "kh_uisdk_quote_cancellation_before_pickup_ios".localized
+        static let freeCancellationASAP = "kh_uisdk_freeCancellationASAP_ios".localized
+        static let freeCancellationAndKeyword = "kh_uisdk_quote_cancellation_and_keyword".localized
+        static let freeCancellationBeforeDriverEnRoute = "kh_uisdk_quote_cancellation_before_driver_departure".localized
+        static let freeWaitingTime = "kh_uisdk_filter_free_waiting_time".localized
+        static let feesAndTaxesIncluded = "Tkh_uisdk_fees_and_taxes_included".localized
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/UITexts.swift:435]: typo at the start of the key.

File: KarhooUISDK/Workers/TripInfoUtility.swift [hunk 13]
```diff
@@ -56,17 +56,15 @@ public struct TripInfoUtility {
                                                  .driverEnRoute: UITexts.GenericTripStatus.enRoute,
                                                  .arrived: UITexts.GenericTripStatus.arrived,
                                                  .passengerOnBoard: UITexts.GenericTripStatus.passengerOnBoard,
-                                                 .completed: UITexts.GenericTripStatus.completed,
-                                                 .unknown: UITexts.GenericTripStatus.unkown,
```
Comment (dianakarh) [inline KarhooUISDK/Workers/TripInfoUtility.swift:60]: What happened to the unknown and incomplete texts?

File: KarhooUISDK/Workers/TripInfoUtility.swift [hunk 14]
```diff
@@ -56,17 +56,15 @@ public struct TripInfoUtility {
...
-                                                 .incomplete: UITexts.GenericTripStatus.incomplete]
+                                                 .completed: UITexts.GenericTripStatus.completed]
         return tripStatuses[tripState] ?? ""
     }
 
     public static func longDescription(trip: TripInfo) -> String {
         let requestedState = String(format: UITexts.Trip.tripStatusRequested,
                 trip.fleetInfo.name)
 
-        let tripStatuses: [TripState: String] = [.unknown: UITexts.Trip.tripStatusUnkown,
```
Comment (dianakarh) [inline KarhooUISDK/Workers/TripInfoUtility.swift:69]: Why was this removed?

File: KarhooUISDKTests/TestCases/Common/TripInfoUtilitySpec.swift [hunk 15]
```diff
@@ -157,12 +157,4 @@ final class TripInfoUtilitySpec: KarhooTestCase {
 
         XCTAssert(TripInfoUtility.longDescription(trip: someTrip) == UITexts.Trip.tripStatusPassengerOnboard)
     }
-    
-    /**
-     * Given: Incomplete trip state
-     * Then: the trip info utility should return its respected status as a string
-     */
-    func testIncompleteTripStatusTranslation() {
```
Comment (dianakarh) [inline KarhooUISDKTests/TestCases/Common/TripInfoUtilitySpec.swift:165]: Are we sure we want to delete the test for this status?

File: KarhooUISDK/CommonUI/UITexts.swift (hunk 4 above)
Comment (mionescu95) [reply KarhooUISDK/CommonUI/UITexts.swift:100]: It was something like 'Something went wrong' so I merged them

File: KarhooUISDK/Workers/TripInfoUtility.swift (hunk 14 above)
Comment (mionescu95) [reply KarhooUISDK/Workers/TripInfoUtility.swift:69]: UITexts.Trip.tripStatusUnkown didn't exist so the unknown status was never showing anything and it is not existing on android

**Review by AleksanderWedrychowskiKarhoo — APPROVED** [review]

File: KarhooUISDK/Translations/Base.lproj/Localizable.strings [hunk 16]
```diff
@@ -71,781 +66,989 @@
...
+
+"kh_uisdk_recents_empty" = "No recent results";
+
+"kh_uisdk_register" = "Register";
+
+"kh_uisdk_report_issue" = "Report Issue";
+
+"kh_uisdk_request_car" = "CONFIRM BOOKING";
+
+"kh_uisdK_request_received" = "Booking Received";
```
Comment (BartlomiejSopalaKarhoo) [inline KarhooUISDK/Translations/Base.lproj/Localizable.strings:829]: capitalized K in uisdK, should be uisdk

File: KarhooUISDK/Translations/es.lproj/Localizable.strings [hunk 17]
```diff
@@ -71,781 +66,989 @@
...
 
-/* Used in the checkout component before booking a ride */
-"Text.TermsConditions.BookingFullString" = "Al hacer una reserva, usted acepta las %1$@ | %2$@ así como los %4$@ de %3$@ | %5$@ de %3$@.";
+"kh_uisdk_legal_notice_text" = "";
 
-"Text.TermsConditions.FullString" = "Al %1$@ con %2$@, aceptas las \n %3$@  |   %4$@";
+"kh_uisdk_legal_notice_title" = "";
 
-"Text.TermsConditions.KarhooPolicyLink" = "https://cdn.karhoo.com/s/mobile/sdk/terms/privacy-policy-sdk.en-gb.html";
+"kh_uisdk_loading_min" = "min";
```
Comment (BartlomiejSopalaKarhoo) [inline KarhooUISDK/Translations/es.lproj/Localizable.strings:591]: In all 5 languages value for this key is : "~ min". Shouldn't be the same here?

File: KarhooUISDK/Translations/Base.lproj/Localizable.strings [hunk 18]
```diff
@@ -2,7 +2,8 @@
 
 "adyen.submitButton.formatted" = "Pre-authorise %@";
 
-/* English: Error codes */
+"continue_button" = "Next";
```
[the remark this answers is not in the document]
Comment (dianakarh) [reply KarhooUISDK/Translations/Base.lproj/Localizable.strings:5]: This is overriding a key from adyen

**Review by dianakarh — APPROVED** [review]
